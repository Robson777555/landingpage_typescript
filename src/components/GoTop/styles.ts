import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;
    box-shadow: 0 2px 5px rgba(0,0,0,0.2);

    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
      box-shadow: 0 4px 8px rgba(0,0,0,0.3);
    }

    &:focus {
      outline: 2px solid ${theme.colors.secondaryColor};
      outline-offset: 2px;
    }

    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;