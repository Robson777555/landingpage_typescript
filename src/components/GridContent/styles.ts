// src/components/GridContent/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;

export const HtmlContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    margin: ${theme.spacings.xhuge} 0;
    text-align: left;

    & > * {
      margin-bottom: ${theme.spacings.medium};
    }

    h1, h2, h3, h4, h5, h6 {
      margin: ${theme.spacings.large} 0;
    }

    pre {
      display: inline-block;
      margin: 0 auto;
      background: rgba(0,0,0,0.1);
      padding: ${theme.spacings.small};
      border-radius: ${theme.spacings.xsmall};
      overflow-x: auto;
      max-width: 100%;
    }

    blockquote {
      border-left: 3px solid ${theme.colors.primaryColor};
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      color: ${theme.colors.mediumGray};
    }

    a {
      color: ${theme.colors.primaryColor};
      text-decoration: none;
      &:hover {
        text-decoration: underline;
      }
    }
  `}
`;