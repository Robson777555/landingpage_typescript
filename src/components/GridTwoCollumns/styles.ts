// src/components/GridTwoCollumns/styles.js
import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: space-between;
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      flex-direction: column;
      text-align: center;
      justify-content: center;
      gap: ${theme.spacings.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.medium};

    /* Estilos para mobile */
    @media ${theme.media.lteMedium} {
      align-items: center;
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    padding: ${theme.spacings.medium};
    max-width: 400px;
    width: 100%;

    img, .next-image {
      max-width: 100%;
      height: auto;
      width: 100%;
      object-fit: contain;
    }

    @media ${theme.media.lteMedium} {
      padding: ${theme.spacings.large} 0;
      max-width: 300px;
    }
  `}
`;

export const Title = styled.h2`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
