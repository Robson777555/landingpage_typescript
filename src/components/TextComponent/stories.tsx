import { Meta, StoryObj } from "@storybook/react";
import { TextComponent } from "./TextComponent";

const meta: Meta<typeof TextComponent> = {
  title: "Components/TextComponent",
  component: TextComponent,
  parameters: {
    layout: "padded",
  },
  argTypes: {
    children: {
      control: "text",
      description: "Text content (supports HTML)",
    },
    size: {
      control: "select",
      options: ["small", "medium", "large"],
    },
    as: {
      control: "select",
      options: ["div", "p", "span"],
    },
  },
  args: {
    children: `Lorem ipsum dolor sit amet consectetur adipisicing elit.
    Nihil unde quod veritatis quam, exercitationem odit totam.`,
    size: "medium",
    as: "div",
  },
};

export default meta;

type Story = StoryObj<typeof TextComponent>;

export const Default: Story = {};

export const SmallText: Story = {
  args: {
    size: "small",
  },
};

export const LargeText: Story = {
  args: {
    size: "large",
  },
};

export const Paragraph: Story = {
  args: {
    as: "p",
  },
};

export const WithHTML: Story = {
  args: {
    children: 'Text with <a href="#">link</a> and <strong>strong</strong> text',
  },
};
