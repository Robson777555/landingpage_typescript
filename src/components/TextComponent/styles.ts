import styled, { css, DefaultTheme } from 'styled-components';

type Size = 'small' | 'medium' | 'large';

const sizes = {
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.small};
    line-height: 1.4;
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.medium};
    line-height: 1.5;
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.fonts.sizes.large};
    line-height: 1.6;
  `,
};

interface ContainerProps {
  $size: Size;
}

export const Container = styled.div<ContainerProps>`
  ${({ theme, $size }) => css`
    font-family: ${theme.fonts.family.default};
    color: ${theme.colors.primaryColor};
    margin: ${theme.spacings.small} 0;

    ${sizes[$size](theme)};

    a {
      color: ${theme.colors.secondaryColor};
      text-decoration: none;
      transition: color 0.3s ease;

      &:hover {
        color: ${theme.colors.primaryColor};
        text-decoration: underline;
      }
    }

    @media ${theme.media.lteMedium} {
      font-size: ${theme.fonts.sizes.small};
    }
  `}
`;