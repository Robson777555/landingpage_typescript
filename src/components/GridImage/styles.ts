// src/components/GridImage/styles.js
import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.large};
      justify-items: center;
    }
  `}
`;

export const GridElement = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  ${({ theme }) => css`
    border-radius: ${theme.spacings.small};
    overflow: hidden;
    box-shadow: 0 4px 8px rgba(0,0,0,0.1);
    transition: transform 300ms ease-in-out;
    width: 100%;
    max-width: 340px;
    margin: 0 auto;
    padding: ${theme.spacings.medium};

    &:hover {
      transform: translateY(-5px);
    }

    img, .next-image {
      max-width: 220px;
      width: 100%;
      height: auto;
      margin: 0 auto;
      display: block;
    }

    h2, h3, h4, h5, h6 {
      font-size: 2.4rem;
      font-weight: bold;
      text-align: center;
      white-space: normal;
      word-break: break-word;
      width: 100%;
      line-height: 1.2;
      margin-bottom: ${theme.spacings.small};
    }

    @media (max-width: 768px) {
      padding: ${theme.spacings.small};
      max-width: 98vw;
      min-width: 0;
      h2, h3, h4, h5, h6 {
        font-size: 2rem;
        font-weight: bold;
        text-align: center;
        white-space: normal;
        word-break: break-word;
        width: 100%;
        line-height: 1.2;
      }
    }
  `}
`;
