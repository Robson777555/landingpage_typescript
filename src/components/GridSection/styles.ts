import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.xlarge};

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      text-align: center;
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    margin-top: ${theme.spacings.large};

    @media (max-width: 768px) {
      grid-template-columns: 1fr;
      gap: ${theme.spacings.xlarge};
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    padding: ${theme.spacings.medium};
    background: ${theme.colors.white};
    border-radius: ${theme.spacings.small};
    box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
    overflow: hidden;

    &:hover {
      transform: translateY(-5px);
      box-shadow: 0 4px 8px rgba(0, 0, 0, 0.15);
    }

    ${HeadingContainer} {
      position: relative;
      left: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: 2.4rem;
      text-align: center;
      word-break: break-word;
      white-space: normal;
      font-weight: bold;
      @media (max-width: 768px) {
        font-size: 2rem;
      }
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 4rem;
      top: -2rem;
      left: 0.5rem;
      transform: rotate(0deg);
      color: ${theme.colors.mediumGray};
      opacity: 0.3;
      z-index: 0;
      @media (max-width: 768px) {
        font-size: 2.5rem;
        top: -1.2rem;
        left: 0.5rem;
      }
    }

    ${TextComponent} {
      text-align: left;
      font-size: 1.2rem;
      @media (max-width: 768px) {
        font-size: 1rem;
      }
    }
  `}
`;

export const Title = styled.h3`
  font-size: 2.4rem;
  font-weight: bold;
  text-align: center;
  word-break: break-word;
  white-space: normal;
  width: 100%;
  line-height: 1.2;
  margin-bottom: ${({ theme }) => theme.spacings.medium};

  @media (max-width: 768px) {
    font-size: 2rem;
  }
`;
