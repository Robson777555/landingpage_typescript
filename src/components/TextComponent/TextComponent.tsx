'use client';
import React from 'react';
import * as Styled from './styles';

type TextSize = 'small' | 'medium' | 'large';
type TextElement = 'div' | 'p' | 'span';

interface TextComponentProps {
  children: string;
  size?: TextSize;
  as?: TextElement;
  className?: string;
}

export const TextComponent = ({
  children,
  size = 'medium',
  as = 'div',
  className = ''
}: TextComponentProps) => {
  return (
    <Styled.Container
      as={as}
      $size={size}
      className={className}
      dangerouslySetInnerHTML={{ __html: children }}
      data-testid="text-component"
    />
  );
};