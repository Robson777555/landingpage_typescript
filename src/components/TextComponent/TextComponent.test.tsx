import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { TextComponent } from './TextComponent';
import { theme } from '../../styles/theme';
import { describe, it, expect } from 'vitest';

describe('<TextComponent />', () => {
  const renderTextComponent = (props = {}) => {
    const defaultProps = {
      children: 'Test Content',
    };

    return render(
      <ThemeProvider theme={theme}>
        <TextComponent {...defaultProps} {...props} />
      </ThemeProvider>
    );
  };

  it('should render text content correctly', () => {
    renderTextComponent();
    expect(screen.getByTestId('text-component')).toBeInTheDocument();
    expect(screen.getByTestId('text-component').textContent).toBe('Test Content');
  });

  it('should render with HTML content', () => {
    renderTextComponent({ children: '<p>HTML <strong>content</strong></p>' });
    expect(screen.getByText('HTML')).toBeInTheDocument();
    expect(screen.getByText('content').tagName).toBe('STRONG');
  });

  it('should apply correct size styles', () => {
    renderTextComponent({ size: 'large' });
    const component = screen.getByTestId('text-component');
    expect(component).toHaveStyle({
      fontSize: theme.fonts.sizes.large,
    });
  });

  it('should render as different HTML element', () => {
    const { container } = renderTextComponent({ as: 'p' });
    expect(container.querySelector('p')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTextComponent();
    expect(container.firstChild).toMatchSnapshot();
  });
});