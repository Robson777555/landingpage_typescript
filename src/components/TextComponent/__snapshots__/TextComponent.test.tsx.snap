// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TextComponent /> > should match snapshot 1`] = `
<div
  class="sc-bXdtCk hmBTGH"
  data-testid="text-component"
>
  Test Content
</div>
`;
